Composer/packages/client/src/components/ImportModal/ImportModal.tsx

```diff
@@ -0,0 +1,338 @@
+  const [importSource, setImportSource] = useState<string>('');
+  const [importPayload, setImportPayload] = useState<ImportPayload>({ name: '', description: '' });
+  const [importedProjectInfo, setImportedProjectInfo] = useState<ImportedProjectInfo | undefined>(undefined);
+  const [modalState, setModalState] = useState<ImportModalState>('connecting');
+  const [existingProject, setExistingProject] = useState<ExistingProjectInfo | undefined>(undefined);
+  const [error, setError] = useState<any>(undefined);
+  const [backupLocation, setBackupLocation] = useState<string>('');
+  const { addNotification } = useRecoilValue(dispatcherState);
+
+  const importAsNewProject = useCallback((info: ImportedProjectInfo) => {
+    // navigate to creation flow with template selected
+    const { alias, description, eTag, name, source, templateDir, urlSuffix } = info;
+    let creationUrl = `/projects/create/${encodeURIComponent(source)}?imported=true&templateDir=${encodeURIComponent(
+      templateDir
+    )}&eTag=${encodeURIComponent(eTag)}`;
+
+    if (name) {
+      creationUrl += `&name=${encodeURIComponent(name)}`;
+    }
+    if (description) {
+      creationUrl += `&description=${encodeURIComponent(description)}`;
+    }
+    if (urlSuffix) {
+      creationUrl += `&urlSuffix=${encodeURIComponent(urlSuffix)}`;
+    }
+    if (alias) {
+      creationUrl += `&alias=${encodeURIComponent(alias)}`;
+    }
+
+    navigate(creationUrl);
+  }, []);
+
+  const importToExistingProject = useCallback(() => {
+    if (importedProjectInfo && existingProject) {
+      setModalState('copyingContent');
+
+      const backupAndSave = async () => {
+        try {
+          // call server to do backup and then save to existing project
+          let res = await fetch(`/api/projects/${existingProject.id}/backup`, { method: 'POST' });
+          const { path } = await res.json();
+          setBackupLocation(path);
+
+          const { eTag, templateDir } = importedProjectInfo;
+          res = await fetch(`/api/projects/${existingProject.id}/copyTemplateToExisting`, {
+            method: 'POST',
+            body: JSON.stringify({ eTag, templateDir }),
+            headers: { 'Content-Type': 'application/json' },
+          });
+
+          // open project and create a notification saying that import was complete
+          if (res.status === 200) {
+            const notification = createNotification({
+              type: 'success',
+              title: '',
+              onRenderCardContent: ImportSuccessNotification({
+                importedToExisting: true,
+                location: existingProject.location,
+              }),
+            });
+            addNotification(notification);
+            navigate(`/bot/${existingProject?.id}`);
+          } else {
+            const err = await res.json();
+            throw err;
+          }
+        } catch (e) {
+          console.error('Something went wrong while saving bot to existing project: ', e);
```

Comment:
Do you want to keep this log?

Reply:
I plan to add better error handling to this flow in the case of errors during sign-in (probably a notification). So I'd prefer to keep some sort of log until then.